include/lsst/meas/astrom/ScaledPolynomialTransformFitter.h

```diff
@@ -0,0 +1,275 @@
+ *  and output data points onto to [-1, 1].  In then fits a polynomial
+ *  transform that, when composed with the input scaling transform and the
+ *  inverse of the output scaling transform, maps the input data points to the
+ *  output data points.
+ */
+class ScaledPolynomialTransformFitter {
+public:
+
+    /**
+     *  Initialize a fit from intermediate world coordinates to pixels using
+     *  source/reference matches.
+     *
+     *  @param[in] maxOrder         Maximum polynomial order for the fit.
+     *
+     *  @param[in] matches          Vector of source-reference matches.
+     *                              The centroids and centroid errors from the
+     *                              sources are used, along with the coords
+     *                              from the reference objects.
+     *
+     *  @param[in] initialWcs       Initial WCS, used to transform reference
+     *                              positions to intermediate world coordinates and
+     *                              populate the "ref" field in the data catalog.
+     *
+     *  @param[in] intrinsicScatter Initial intrinsic scatter to be added in
+     *                              quadrature with source measurement errors in
+     *                              setting the uncertainty on each match.
+     *
+     *  This initializes the data catalog with the following fields:
+     *   - ref_id:             reference object ID
+     *   - src_id:             source ID from the match
+     *   - src_{x,y}:          source position in pixels
+     *   - src_{xSigma,ySigma,x_y_Cov}:  uncertainty on source positions,
+     *                         including measurement errors and the inferred
+     *                         intrinsic scatter.
+     *   - intermediate_{x,y}: reference positions in intermediate world
+     *                         coordinates.
+     *   - initial_{x,y}:      reference positions transformed by initialWcs.
+     *   - model_{x,y}:        reference positions transformed by current
+     *                         best-fit distortion.
+     *   - valid:              Flag that is set for objects that have not been
+     *                         rejected as outliers.
+     */
+    static ScaledPolynomialTransformFitter fromMatches(
+        int maxOrder,
+        afw::table::ReferenceMatchVector const & matches,
+        afw::image::Wcs const & initialWcs,
+        double intrinsicScatter
+    );
+
+    /**
+     *  Initialize a fit that inverts an existing transform by evaluating and
+     *  fitting to points on a grid.
+     *
+     *  @param[in] maxOrder         Maximum polynomial order for the fit.
+     *
+     *  @param[in] bbox             Bounding box for the grid in the input
+     *                              coordinates of the toInvert transform
+     *                              (or equivalently the output coordinates
+     *                              of the transform to be fit).
```

Comment:
In the way it's used in `FitSipDistortionTask`, the bounding box has to be in intermediate world coordinates, and given that we only start with the bounding box of the image and we're trying to *find* the mapping from pixel coordinates to intermediate world coordinates here, our only choice is to transform the image bounding box using (part of) the initial guess WCS.

 I don't think it's appropriate to include that description in the docs for `fromGrid`, since it's more general than that, but I will add code comments to `FitSipDistortionTask` where I'm computing the bbox.